controller/src/bin/xcvradm.rs

```diff
@@ -13,29 +20,145 @@ async fn request_handler(_: SpRpcRequest) -> Result<HostRpcResponse, Error> {
     Err(Error::Protocol(MessageError::ProtocolError))
 }
 
+fn parse_xvcr_list(s: &str) -> Result<Vec<u8>, String> {
+    s.split(",")
+        .map(|x| {
+            x.parse()
+                .map_err(|_| String::from("invalid transceiver index"))
+        })
+        .collect()
```

Comment:
We could flip the order of the collect / map_err here, which makes it shorter and point-free (but adds a turbofish); I'm not sure if this is an improvement!
```suggestion
    s.split(",")
        .map(str::parse)
        .collect::<Result<_, _>>()
        .map_err(|_| String::from("invalid transceiver index"))
```

Reply:
That looks a bit nicer to me, thanks!